public/app/plugins/datasource/grafana-azure-monitor-datasource/app_insights/app_insights_datasource.ts

```diff
@@ -20,8 +19,7 @@ export default class AppInsightsDatasource {
   applicationId: string;
   logAnalyticsColumns: { [key: string]: LogAnalyticsColumn[] } = {};
 
-  /** @ngInject */
-  constructor(instanceSettings: DataSourceInstanceSettings<AzureDataSourceJsonData>, private templateSrv: TemplateSrv) {
+  constructor(instanceSettings: DataSourceInstanceSettings<AzureDataSourceJsonData>) {
```

Comment:
I made this proposal https://github.com/grafana/grafana/pull/23048 where we will start moving Angular services into dependencies. This works too, but the drawback is that you have to use the global jest.mock in your tests and that offers very little control over the mock as it can only be defined once per file.

Reply:
@hugohaggmark I think we could turn your proposal into a guideline in the front-end style guide. I remember some people liked that proposal (me included) so having it as a "formal" guide would help to push people into this style. WDYT?

Reply:
@dprokop I agree and will add it :)

Reply:
Opened https://github.com/grafana/grafana/pull/25391 @dprokop